daemon/container_operations.go

```diff
@@ -678,6 +791,13 @@ func disconnectFromNetwork(container *container.Container, n libnetwork.Network,
 	}
 
 	delete(container.NetworkSettings.Networks, n.Name())
+
+	if daemon.clusterProvider != nil && n.Info().Dynamic() && !container.Managed {
```

Comment:
If `n` is non-nil `n.Info` will always be non-nil as well. So don't need to check.